deployment/terraform/examples/aws-docker-swarm/user-data-swarm.tftpl

```diff
@@ -0,0 +1,36 @@
+#!/bin/bash
+set -xe
+
+# Install Docker
+echo "Installing Docker..."
+sudo apt-get update
+sudo apt-get install docker -y docker.io
+echo "Installing aws cli..."
+sudo apt-get install awscli -y
+sudo usermod -aG docker ubuntu 
+sudo systemctl enable docker
+sudo systemctl start docker
+
+echo ${node_number}
+
+if [[ "${node_number}" == "0" ]]; then
+    # Initialize swarm and generate tokens
+    echo "Initializing Swarm..."
+    docker swarm init --advertise-addr $(hostname -i):2377
+    export MANAGER_TOKEN=$(docker swarm join-token manager | grep token | xargs echo -n)
+    export WORKER_TOKEN=$(docker swarm join-token worker | grep token | xargs echo -n)
+
+    # Store tokens securely (e.g., AWS SSM Parameter Store)
+    echo "Storing Manager Token [$MANAGER_TOKEN]"
+    echo "Storing Worker Token [$WORKER_TOKEN]"
+    aws ssm put-parameter --region us-east-2 --name '/swarm/token/manager' --type String --overwrite --value "$MANAGER_TOKEN"
```

Comment:
we may comment here to be secure

`Devs  can specify a KMS key with aws ssm put-parameter --key-id <KMS_KEY_ID>, ensuring encrypted storage`

Reply:
Yeah, I will add the comment.
This will eventually get re-worked to use another method (HTTP endpoint?) that is more cloud agnostic.